crates/shared/src/bad_token/cache.rs

```diff
@@ -55,22 +64,84 @@ impl CachingDetector {
             .unwrap()
             .insert(token, (Instant::now(), quality));
     }
+
+    fn insert_many_into_cache(&self, tokens: impl Iterator<Item = (H160, TokenQuality)>) {
+        let mut cache = self.cache.lock().unwrap();
+        for (token, quality) in tokens {
+            cache.insert(token, (Instant::now(), quality));
```

Comment:
nit: Let's compute `Instant::now()` once outside the loop and copy the same value.
Doesn't really matter if the timestamps are different by nanoseconds.